### PR TITLE
fix(eslint-plugin): [no-unsafe-member-access] ignore MemberExpression's whose parents are either TSClassImplements or TSInterfaceHeritage

### LINKED ISSUE #2740: [no-unsafe-member-access] Extending namespaces trigger error
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
```TS
declare namespace FG {
	interface A {}
}

interface B extends FG.A {
}
```
You can also look at this example repo: https://github.com/susnux/typescript-eslint-no-unsafe-member-access

**Expected Result**
No error message

**Actual Result**

> typescript-eslint-no-unsafe-member-access/src/test.d.ts
>  6:21  error  Unsafe member access .A on an any value  @typescript-eslint/no-unsafe-member-access

**Additional Info**
Nothing special, you can simply reproduce this.
I experienced the same issue (this) as reported by @tao-cumplido in #1868.

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | `2.27.0` |
| `@typescript-eslint/parser`             | `2.27.0 |
| `@typescript-eslint/typescript-estree`  | `2.27.0` |
| `@typescript-eslint/experimental-utils` | `2.27.0` |
| `TypeScript`                            | `4.0.5` |
| `node`                                  | `12.18.4` |
| `npm`                                   | `6.14.6` |

### PR DESCRIPTION
**Pull request by hsmitty93**
Fixes #2740 

- I feel like the selector is too long to be on one line, but it wouldn't work if I split it. Any recommendations to shorten it? Or is it good to go like it is.

### CODE REVIEW

File: packages/eslint-plugin/src/rules/no-unsafe-member-access.ts [hunk 1]
```diff
@@ -71,7 +71,8 @@ export default util.createRule({
     }
 
     return {
-      MemberExpression: checkMemberExpression,
+      // ignore MemberExpression if it's parent is TSClassImplements or TSInterfaceHeritage
+      'MemberExpression:not(TSClassImplements > MemberExpression):not(TSInterfaceHeritage > MemberExpression)': checkMemberExpression,
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/no-unsafe-member-access.ts:75]: `:not(...)` accepts any valid selector as its argument, which means I believe you can simplify this slightly by merging everything into one:
```suggestion
      ':not(TSClassImplements, TSInterfaceHeritage) > MemberExpression': checkMemberExpression,
```

**Review by bradzacher — CHANGES_REQUESTED** [review]
great work! the idea and tests LGTM.

But I think we can make the selector a little easier on the eyes :)

File: packages/eslint-plugin/src/rules/no-unsafe-member-access.ts (hunk 1 above)
Comment (hsmitty93) [reply packages/eslint-plugin/src/rules/no-unsafe-member-access.ts]: Nice, that is simpler and tests still pass. I've added the change.

**Review by bradzacher — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**